### PR TITLE
Check Options in Specifications (WIP)

### LINKED ISSUE #23: Set check options in specification code
I'm just going through the project with a fresh pair of eyes, thinking about how a new user would interact with Quickstrom. Here's what I'm wondering:

When looking at the examples the tests are ran using the cli, but I wonder if some of the attributes passed to the cli can in fact be in the tests/specs? eg: 
https://github.com/quickstrom/quickstrom/blob/9e74c0da944799ae9d3520dc67c576c1fbd8c7e8/specs/run-all.sh#L49
can `http://todomvc.com/examples/vue/` be set as part of a top level spec file, or some sort of a runner from within the specs?

When looking at projects like Puppeteer, you can have full control of the test runner from within the specs. (ignoring it's JS!)

```JS
const puppeteer = require('puppeteer');

(async () => {
  const browser = await puppeteer.launch()
  const page = await browser.newPage()
  await page.goto('https://www.google.com/') // this line telling test runner where to go
  page.on('dialog', async dialog => {
    console.log(dialog.message())
    await dialog.dismiss()
  })
  await page.evaluate(() => alert('This message is inside an alert box'))
  await browser.close()
})()
```

Even if it's not inline as part of a spec file, but could be a top level runner:
```Purescript
module Runner where

import Quickstrom (Options(..), Flags(..), Results(..), runQuickStromIO)
import Spec1 (spec1)
import Spec2 (spec2)


main = do
  runQuickStromIO [ { spec: spec1
                    , loc: "../MultiPage.html"
                    , opts: [Options]
                    , flags: [Flags]
                    , res: Result
                  ]
  runQuickStromIO [ { spec: spec2
                    , loc: "http://www.webA.com"
                    , opts:[Options]
                    , flags: [Flags]
                    , res: Result 
                    }
                  , { spec: spec2
                    , loc:  "http://www.webB.com" 
                    , opts: [Options] 
                    , flags: [Flags]
                    , res: Result
                    }
                  ]
```

Ignoring my terrible naming conventions and if you'd even use records, but the idea being you have control of what the test runner is doing from within the code you are writing the specs/test.

Under the hood it would probably just be using the cli directly and from there could control styling of outputs etc.

This would help with context switching, because how likes to write shell scripts hehe

### PR DESCRIPTION
**Pull request by Cmdv**
fixes #23

WIP, need some pointers in regards to how far I make things `Maybe`'s

### CODE REVIEW

File: runner/src/Quickstrom/Run.hs [hunk 1]
```diff
@@ -298,7 +280,7 @@ runActions' spec = do
 
 data Shrink a = Shrink Int a
 
-shrinkForest :: (a -> [a]) -> Int -> a -> Forest (Shrink a)
+shrinkForest :: (a -> [a]) -> Maybe Int -> a -> Forest (Shrink a)
```
Comment (Cmdv) [inline runner/src/Quickstrom/Run.hs:283]: Bit weary about making this into a `Maybe`!

File: runner/src/Quickstrom/Options.hs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+module Quickstrom.Options where
+
+import qualified Data.Aeson as JSON
+import GHC.Generics (Generic)
+import Quickstrom.Prelude hiding (catch, check, trace)
+import Quickstrom.WebDriver.Class
+import Text.URI
+
+data CheckOptions f = CheckOptions
```
Comment (Cmdv) [inline runner/src/Quickstrom/Options.hs:12]: Moved this to it's own file, as I was getting cyclic issues

**Comment by Cmdv** [comment]
@owickstrom I've tried to get things moving but I'm stuck on creating a [parser](https://github.com/quickstrom/quickstrom/pull/50/files#diff-8f1b4a11df5ab7f7c973e15efd6ada5fR230), I know I need to do something similar to what was mentioned inside of slack but I've not managed it.

Any pointers would be greatly appreciated.

**Comment by Cmdv** [comment]
I'm going to shut this down as I imagine it's become outdated 😃

**Comment by owickstrom** [comment]
Yeah it might be now. A lot of changes are coming to the spec language anyway so let's hold off with this change. Thanks for your participation!

**Pull request closed without merging**